### PR TITLE
Enable nullable annotations in ConversionWrappers

### PR DESCRIPTION
**Pull request by BCSharp**

### CODE REVIEW

File: Src/IronPython/Runtime/ConversionWrappers.cs [hunk 1]
```diff
@@ -2,20 +2,23 @@
...
 namespace IronPython.Runtime {
 
     public class ListGenericWrapper<T> : IList<T> {
-        private readonly IList<object> _value;
+        private readonly IList<object?> _value;
         // PEP 237: int/long unification (GH #52)
         private static readonly bool IsBigIntWrapper = typeof(T) == typeof(BigInteger) || typeof(T) == typeof(BigInteger?);
 
-        public ListGenericWrapper(IList<object> value) { _value = value; }
+        public ListGenericWrapper([NotNone] IList<object?> value) { _value = value; }
```
[the remark this answers is not in the document]
Comment (BCSharp) [reply Src/IronPython/Runtime/ConversionWrappers.cs:21]: > Do we interact with these wrappers via Python

Not that I know of.

> does `NotNone` have any effect?

Yes, as usual, it prevents `None` from binding:

```pycon
>>> import clr
>>> clr.AddReference('IronPython')
>>> import System
>>> import IronPython
>>> IronPython.Runtime.ListGenericWrapper[System.Int32](None)  
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: expected IList[object], got NoneType
```

What surprised me though is that one of the conversion wrappers, `IEnumerableOfTWrapper` has `[PythonType("enumerable_wrapper")]` as if it was intended to be used from Python, but I can't find it anywhere under that name.

[the remark this answers is not in the document]
Comment (slozier) [reply Src/IronPython/Runtime/ConversionWrappers.cs:21]: > > does `NotNone` have any effect?
> 
> Yes, as usual, it prevents `None` from binding:

I'd also be fine with just throwing an `ArgumentNullException` in the constructors.

> What surprised me though is that one of the conversion wrappers, `IEnumerableOfTWrapper` has `[PythonType("enumerable_wrapper")]` as if it was intended to be used from Python, but I can't find it anywhere under that name.

Yeah, saw that as well. Unfortunately it predates git history so it's hard to say what the idea was. Could probably zap it...

[the remark this answers is not in the document]
Comment (BCSharp) [reply Src/IronPython/Runtime/ConversionWrappers.cs:21]: > Yeah, saw that as well. Unfortunately it predates git history so it's hard to say what the idea was. Could probably zap it...

OK, zap it is.

**Review by slozier — APPROVED** [review]
Looks good to me!

**Pull request merged**